public/app/features/scopes/internal/utils.ts

```diff
@@ -43,3 +43,88 @@ export function getScopesFromSelectedScopes(scopes: SelectedScope[]): Scope[] {
 export function getScopeNamesFromSelectedScopes(scopes: SelectedScope[]): string[] {
   return scopes.map(({ scope }) => scope.metadata.name);
 }
+
+export function groupDashboards(dashboards: ScopeDashboardBinding[]): SuggestedDashboardsFoldersMap {
```

Comment:
Nit about this utils, as I see they are indirectly tested through UI tests. I think we should also include unit tests for these so that its easy to find them and reason about them. The scopes UI tests doesn't make it obvious these core logic lives here.

Reply:
Added tests for both grouping and filtering